bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/services/UrlProviderImpl.java

```diff
@@ -149,6 +158,69 @@ protected void deactivate() {
         newCategoryUrlFormat = null;
     }
 
+    private ProductUrlFormat getProductFormatUrlFormat(SlingHttpServletRequest request, Page page) {
+        if (request == null && page == null) {
+            throw new IllegalArgumentException("The request and the page parameters cannot both be null");
+        }
+
+        ProductUrlFormat productUrlFormat = newProductUrlFormat;
+        Resource resource = request != null ? request.getResource() : page.getContentResource();
```

Comment:
Prefer the `page.getContentResource()` over the `request.getResource()`.

The page is (implicitly) always in the context of the `currentPage` (initially requested page). The resource may be somewhere else due to an include. For XF for example the request.getResource() may point to a different CA Configuration than the page original  request. 

If page != null, use page.getContentResource
else if request != null use request.getResource
else throw IAE